storey/writers.py

```diff
@@ -42,16 +51,30 @@ async def _do(self, event):
         try:
             if not self._open_file:
                 self._open_file = await aiofiles.open(self._path, mode='w')
+            data = event.body
+            if isinstance(data, dict):
+                if self._first_event and not self._columns:
+                    self._columns = list(data.keys())
+                    self._columns.sort()
+                if self._columns:
```

Comment:
won't you always have columns by this line?

Reply:
Not if we get mixed events (list/dict). Funky usage but it's up to users.